### ISSUE TITLE
v0.14.7 huge hash performance degradation

### ISSUE THREAD

**Issue by ssergiienko** [syncthing/syncthing#3614]
v0.14.6 reports this:
```
23:36:13 INFO: syncthing v0.14.6 "Dysprosium Dragonfly" (go1.7 linux-amd64) builduser@svetlemodry 2016-09-06 15:26:27 UTC
23:36:13 INFO: Single thread hash performance is ~186 MB/s
```

 v0.14.7 consistently reports:
```
23:35:03 INFO: syncthing v0.14.7 "Dysprosium Dragonfly" (go1.7.1 linux-amd64) builduser@svetlemodry 2016-09-21 18:31:42 UTC
23:35:03 INFO: Single thread hash performance is ~43 MB/s
```
I've tried to restart it few times and every time it was around ~40MB/s

OS: Archlinux
CPU:
```
Architecture:          x86_64                                                                                                                                                                
CPU op-mode(s):        32-bit, 64-bit                                                                                                                                                        
Byte Order:            Little Endian                                                                                                                                                         
CPU(s):                8                                                                                                                                                                     
On-line CPU(s) list:   0-7                                                                                                                                                                   
Thread(s) per core:    2                                                                                                                                                                     
Core(s) per socket:    4                                                                                                                                                                     
Socket(s):             1                                                                                                                                                                     
NUMA node(s):          1                                                                                                                                                                     
Vendor ID:             GenuineIntel                                                                                                                                                          
CPU family:            6                                                                                                                                                                     
Model:                 42                                                                                                                                                                    
Model name:            Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz                                                                                                                              
Stepping:              7                                                                                                                                                                     
CPU MHz:               1863.525                                                                                                                                                              
CPU max MHz:           5900.0000                                                                                                                                                             
CPU min MHz:           1600.0000                                                                                                                                                             
BogoMIPS:              6787.48                                                                                                                                                               
Virtualization:        VT-x                                                                                                                                                                  
L1d cache:             32K                                                                                                                                                                   
L1i cache:             32K                                                                                                                                                                   
L2 cache:              256K                                                                                                                                                                  
L3 cache:              8192K                                                                                                                                                                 
NUMA node0 CPU(s):     0-7
Flags:                 fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc aperfmperf eagerfpu pni pclmulqdq dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm pcid sse4_1 sse4_2 popcnt tsc_deadline_timer aes xsave avx lahf_lm epb tpr_shadow vnmi flexpriority ept vpid xsaveopt dtherm ida arat pln pts
```

**Comment by lkwg82** [comment]
Do you have some aggressive power saving cpu governor activated?

I notice your CPU is at 1,8Ghz, so this is maybe the cause? Is this a Laptop?

**Comment by ssergiienko** [comment]
As far as I know - I don't) Nothing has changed in configuration for a years. It's not a laptop, it's a desktop with an overclocked CPU. But even if I would have it, why v0.14.6 is not affected in any way?
I've measured both version results one after another and result is stable.
Or am I missing something?

I found this in release notes:
> Improvements in this release include faster (i.e., less CPU intensive) hashing on some Intel and ARM64 CPU:s

Can this changes cause this problem?
